source/common/http/http2/codec_impl.cc

```diff
@@ -1070,25 +1107,40 @@ void ConnectionImpl::sendSettings(
   }
 }
 
+int ConnectionImpl::setAndCheckNghttp2CallbackStatus(Status&& status) {
+  nghttp2_callback_status_ = std::move(status);
```

Comment:
ASSERT that the status was ok in case of overwriting

Reply:
I'm unsure we can safely ASSERT this. I'm thinking that it might be possible that sendPendingFrames is called twice, i.e. by filters. I think the question is whether we want to keep the first error status or the last.